app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDateTimePicker.kt

```diff
@@ -0,0 +1,226 @@
+            onDismissRequest()
+          },
+      )
+  MyAppTheme {
+    CalendarDialog(
+        state = useCaseState,
+        selection =
+            CalendarSelection.Date(
+                onSelectDate = { localDate ->
+                  // Trigger onDateSelected only when a date is selected
+                  onDateSelected(localDate.year, localDate.monthValue, localDate.dayOfMonth)
+                  onFinishedRequest()
+                },
+                selectedDate = LocalDate.now(),
+                onNegativeClick = { onDismissRequest() }),
+        config =
+            CalendarConfig(
+                locale = Locale.getDefault(),
+                style = CalendarStyle.WEEK,
+                cameraDate = LocalDate.now(),
+                yearSelection = true,
+                monthSelection = true,
+                disabledDates = disableDates),
+        header = Header.Default("Select Date"))
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TimePickerDialog(
+    onDismissRequest: () -> Unit,
+    onTimeSelected: (Int, Int) -> Unit,
+    timeViewTest: Boolean
+) {
+  val timePickerState =
+      rememberTimePickerState(
+          initialHour = LocalTime.now().hour,
+          initialMinute = LocalTime.now().minute,
+          is24Hour = true)
+
+  val displayModeState = remember { mutableStateOf(DisplayMode.Input) }
+
+  AlertDialog(
+      onDismissRequest = { onDismissRequest() },
+      confirmButton = {
+        TextButton(onClick = { onTimeSelected(timePickerState.hour, timePickerState.minute) }) {
+          Text(
+              "OK",
+              fontSize = 16.sp,
+          )
+        }
+      },
+      dismissButton = {
+        TextButton(onClick = { onDismissRequest() }) {
+          Text(
+              "Cancel",
+              fontSize = 16.sp,
+          )
+        }
+      },
+      title = {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier.fillMaxWidth()) {
+              Text(
+                  text = "Select Time",
+              )
+              DisplayModeToggleButton(
+                  displayModeState = displayModeState, modifier = Modifier.padding(start = 8.dp))
+            }
+      },
+      text = {
+        Column {
+          if (displayModeState.value == DisplayMode.Picker && timeViewTest) {
+            TimeInput(
+                state = timePickerState,
+                modifier = Modifier.fillMaxWidth(),
+                colors =
+                    TimePickerDefaults.colors(
+                        timeSelectorSelectedContainerColor = colorScheme.primary.copy(alpha = 0.3f),
+                        timeSelectorSelectedContentColor = colorScheme.onBackground,
+                        clockDialColor = colorScheme.primary.copy(alpha = 0.1f),
+                        timeSelectorUnselectedContainerColor = Color(0xffe9e0e0),
```

Comment:
It's always nice to change the hard coded color to a value or something but not a must change.

Reply:
Got it !